### PR TITLE
Added fields to Chat

### PR DESCRIPTION
**Pull request by lyteloli**
# Description
Added a few extensions to Chat type.

## Type of change

Please delete options that are not relevant.

- [x] Documentation (typos, code examples or any documentation update)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Full name test
- [x] Mention test
- [x] User URL test

**Test Configuration**:
* Operating System: MacOS Catalina 0.15.5 (19F96)
* Python version: 3.8.2

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: aiogram/api/types/chat.py [hunk 1]
```diff
@@ -51,3 +52,52 @@ class Chat(TelegramObject):
     """For supergroups, name of group sticker set. Returned only in getChat."""
     can_set_sticker_set: Optional[bool] = None
     """True, if the bot can change the group sticker set. Returned only in getChat."""
+
+    @property
+    def full_name(self) -> Optional[str]:
```
Comment (uwinx) [inline aiogram/api/types/chat.py:57]: maybe `full_title`?

File: aiogram/api/types/chat.py [hunk 2]
```diff
@@ -51,3 +52,52 @@ class Chat(TelegramObject):
...
+        """
+        Get user's full name for private chats and chat title for groups/channels.
+        """
+        if self.type == ChatType.PRIVATE:
+            return f'{self.first_name} {self.last_name}' if self.last_name else self.first_name
+
+        return self.title
+
+    @property
+    def mention(self) -> Optional[str]:
```
Comment (uwinx) [inline aiogram/api/types/chat.py:67]: I think `mention` is pretty evil, we may end up without "mention" at all

File: aiogram/api/types/chat.py [hunk 3]
```diff
@@ -51,3 +52,52 @@ class Chat(TelegramObject):
...
+        """
+        if self.username:
+            return f'@{self.username}'
+        elif self.type == ChatType.PRIVATE:
+            return self.full_name
+
+        return None
+
+    @property
+    def user_url(self) -> str:
```
Comment (uwinx) [inline aiogram/api/types/chat.py:79]: Why `user_url` is in `Chat`? I don't see any use-case

File: tests/test_api/test_types/test_chat.py [hunk 4]
```diff
@@ -35,9 +35,14 @@ def test_mention(self, chat_type: Optional[str], title: Optional[str], first_nam
...
         [
-            ["private", 123124123, 'tg://user?id=123124123']
+            ["private", 123124123, 'tg://user?id=123124123'],
+            ["channel", -10032432324, None]
         ]
     )
     def test_user_url(self, chat_type: Optional[str], chat_id: int, result):
         chat = Chat(type=chat_type, id=chat_id)
-        assert chat.user_url == result
+        if chat_type != 'private':
```
Comment (igorsereda) [inline tests/test_api/test_types/test_chat.py:44]: Unit tests should not contain any logic. This test should be divided into two different cases, positive and negative, with different sets of input data.

File: tests/test_api/test_types/test_chat.py [hunk 5]
```diff
@@ -35,9 +35,14 @@ def test_mention(self, chat_type: Optional[str], title: Optional[str], first_nam
...
+            with pytest.raises(TypeError):
+                assert chat.user_url
```
Comment (igorsereda) [inline tests/test_api/test_types/test_chat.py:46]: I think, the assert keyword is redundant here.

File: aiogram/api/types/chat.py [hunk 6]
```diff
@@ -51,3 +52,52 @@ class Chat(TelegramObject):
...
+
+    @property
+    def full_name(self) -> Optional[str]:
+        """
+        Get user's full name for private chats and chat title for groups/channels.
+        """
+        if self.type == ChatType.PRIVATE:
+            return f'{self.first_name} {self.last_name}' if self.last_name else self.first_name
+
+        return self.title
```
Comment (JrooTJunior) [inline aiogram/api/types/chat.py:64]:
```suggestion
        return cast(str, self.title)
```

**Review by BogdanDevUA — APPROVED** [review]

**Comment by evgfilim1** [comment]
I guess this PR can be closed due to inactivity.
@lyteloli if you wanna improve it, feel free to re-open!

**Pull request closed without merging**